scalyr_agent/util.py

```diff
@@ -677,6 +679,46 @@ def get_web_url_from_upload_url(server):
     return server
 
 
+def parse_data_rate_string(value):
+    """Return a rate in bytes per second parsed from a string of a float or int followed by a unit.
+    Takes into account bit (`b`) vs byte (`B`), ignores capitalization of things like `k` vs `K` and the time unit
+    denominator. Allowed numerators go up to terabytes, and allowed denominators go up to weeks.
+
+    :param value: String with value and unit
+    :return: The parsed rate in bytes per second, or raise ValueError if it could not be parsed
+    """
+    m = re.search(r"(-?\d+\.?\d*)\s*([kKmMgGtT]?)([bB])/([sSmMhHdDwW])", value)
+    if m:
+        value = float(m.group(1))
+        numerator = m.group(2).upper()
+        bit_or_byte = m.group(3)
+        denominator = m.group(4).upper()
+
+        if numerator == "K":
+            value = value * 1024
```

Comment:
For bonus points, you could be in support for MiB vs MB... but I don't think it is really necessary.